### PR TITLE
ZK-4300: Choosing Jan 1st, 1900 results in wrong year in format yy Da…

### PR DESCRIPTION
**Pull request by scribetw**
…tebox

### CODE REVIEW

File: zktest/test/java/org/zkoss/zktest/zats/test2/B86_ZK_4300Test.java [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+		waitResponse();
+
+		click(datebox.$n("btn"));
+		waitResponse();
+
+		Assert.assertEquals("1900", jq(cal.toWidget().$n("ty")).text());
+		Assert.assertThat(getZKLog(), containsString("1900"));
+		closeZKLog();
+	}
+}
```
Comment (DevChu) [inline zktest/test/java/org/zkoss/zktest/zats/test2/B86_ZK_4300Test.java]: set animation speed 0

File: zul/src/archive/web/js/zul/db/Datebox.js [hunk 2]
```diff
@@ -368,8 +368,11 @@ zul.db.Datebox = zk.$extends(zul.inp.FormatWidget, {
 			}
 		}
 		if (val) {
-			var format = this.getFormat(),
-				d = new zk.fmt.Calendar().parseDate(val, pattern || format, !this._lenient, this._value, this._localizedSymbols, tz, this._strictDate);
+			// We cannot use this._value in this case, which won't trigger onChange
+			// event. Using clone date instead.
+			var date = this._value ? Dates.newInstance(this._value.getTime(), tz) : null,
```
Comment (DevChu) [inline zul/src/archive/web/js/zul/db/Datebox.js]: remove tz

File: zktest/test/java/org/zkoss/zktest/zats/test2/B86_ZK_4300Test.java (hunk 1 above)
Comment (scribetw) [reply zktest/test/java/org/zkoss/zktest/zats/test2/B86_ZK_4300Test.java]: https://tracker.zkoss.org/browse/ZK-4308

**Comment by scribetw** [comment]
@zkadmin rebase

**Comment by zkadmin** [comment]
I just pushed up the changes, enjoy!

**Pull request merged**